pulp_ostree/tests/functional/conftest.py

```diff
@@ -0,0 +1,121 @@
+    _api_client_set.add(api_client)
+    yield api_client
+    _api_client_set.remove(api_client)
+
+
+@pytest.fixture()
+def ostree_content_refs_api(ostree_client):
+    """Fixture that returns an instance of ContentRefsApi"""
+    return ContentRefsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RepositoriesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_remotes_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RemotesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_versions_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeVersionsApi"""
+    return RepositoriesOstreeVersionsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repository_sync_url(ostree_client):
+    """Fixture that returns an instance of RepositorySyncURL"""
+    return RepositorySyncURL(ostree_client)
+
+
+@pytest.fixture()
+def ostree_content_commits_api(ostree_client):
+    """Fixture that returns an instance of ContentCommitsApi"""
+    return ContentCommitsApi(ostree_client)
+
+
+@pytest.fixture()
+def remote(ostree_remotes_api):
+    """Fixture that creates an ostree Remote and returns the object created"""
+    body = gen_ostree_remote(depth=0, policy="immediate")
+    return ostree_remotes_api.create(body)
+
+
+@pytest.fixture()
+def repo(ostree_repositories_api):
+    """Fixture that creates an ostree Repository"""
+    return ostree_repositories_api.create(gen_repo())
+
+
+@pytest.fixture()
+def ostree_repository_version(
+    remote, repo, ostree_repositories_api, ostree_repositories_versions_api
+):
+    """
+    Fixture that syncs a Remote ostree Repository,
+    waits until the sync task completes and returns the
+    created repository version object.
+    """
+    repository_sync_data = RepositorySyncURL(remote=remote.pulp_href)
+    response = ostree_repositories_api.sync(repo.pulp_href, repository_sync_data)
+    repo_version = monitor_task(response.task).created_resources[0]
+
+    return ostree_repositories_versions_api.read(repo_version)
+
+
+@pytest.fixture()
+def ostree_repositories_delete(ostree_repositories_api, repo):
+    """Fixture that deletes an ostree Repository"""
+    return ostree_repositories_api.delete(repo.pulp_href)
+
+
+@pytest.fixture()
+def ostree_remotes_delete(ostree_remotes_api, remote):
+    """Fixture that deletes an ostree Remote"""
+    return ostree_remotes_api.delete(remote.pulp_href)
+
+
+@pytest.fixture()
+def setup_filter(
+    repo,
+    remote,
+    ostree_repository_version,
+    ostree_repositories_api,
+    ostree_remotes_api,
+):
+    """Setup for ostree filter tests"""
+    delete_orphans()
+
+    yield ostree_repository_version
+    ostree_repositories_api.delete(repo.pulp_href)
+    ostree_remotes_api.delete(remote.pulp_href)
+    delete_orphans()
```

Comment:
If this is a special setup code for a few tests in one module, the fixture should live there too.
Instead of calling delete_orphans, use the delete_orphans_pre fixture (I always suppose to check if it's needed at all...).